eng/common/scripts/Helpers/Lock-Helpers.ps1

```diff
@@ -0,0 +1,17 @@
+function FileLockEnter([string]$FilePath) {

```

Comment:
I don't like this approach as it has race conditions itself so if the goal is to prevent stomping on each other then we need a higher level step. Perhaps we only do one npm install and then disable the install on all the other parallel jobs.

Reply:
I'm also curious whether the race condition is an actual problem that will occur as a result of other automation, or rather a potential problem we anticipate if users run multiple npm installs in parallel?

Reply:
If we did that @pshao25 would option 1 install it locally inside the project folder?  Or is it up to the local user to know not to do things in parallel?

Would they even know what parallel options are magically happening under the covers if they ran
`dotnet build ./eng/service.proj /t:GenerateCode`

Reply:
Couldn't we do the install once up front and then have our scripts have some opt-out, perhaps an env variable, that tells it not to do install any longer and we simply skip all those parallel installs?

Reply:
I prefer command line parameter vs environment.  @weshaggard  is there a specific advantage you are after for using env?

Reply:
It ends up depending on how deep this needs to be passed. Env variables make it easier to set on the outside and not need to worry about the layers. Also I suspect this is going to be a challenge for other languages as well which doesn't have msbuild type properties so we would need some way to do this for them as well.

Reply:
It's not deep at all. Other languages don't have msbuild type properties, but they still could pass the parameter from where they call the script. (It's just we .net call the script from msbuild, so we pass the parameter from msbuild type properties)